feat: Document not authenticated users management (#147)
lenra-io/docs: src/markdown/guides/public-routes.md

---
description: Give access to not authenticated users.
---

Do you want to give access to your app to not authenticated users?
In this guide we will explain how to create a public route and how to use it in a front-end app.

## Create a public route

To create a public route, you need to set the `roles` property of one of your manifest routes to `["guest"]` or `["guest", "user"]` if you want the same route to be accessible to authenticated and not authenticated users.

Here is an example of a public route:

```json
{
	"json": {
		"routes": [
			{
				"path": "/public",
				"view": "myPublicView",
				"roles": ["guest"]
			},
			{
				"path": "/all",
				"view": "myView",
				"roles": ["guest", "user"]
			},
			{
				"path": "/private",
				"view": "myPrivateView"
			}
		]
	}
}
```

## Use a public route in a front-end app

> To use a public as a not authenticated user with the [{:rel="noopener" target="_blank"} Lenra JavaScript client lib](https://github.com/lenra-io/client-lib-js) you just have to use the `openSocket()` method to connect to the Lenra server without token and then open the route:

Reviewer: ```suggestion
To use the [{:rel="noopener" target="_blank"} Lenra JavaScript client lib](https://github.com/lenra-io/client-lib-js) as an unauthenticated user, simply use the `openSocket()` method to connect to the Lenra server without a token, and then open the desired route.
```
